### DISCUSSION TITLE
Welcome to greenbone-gvm-openvas-for-docker Discussions!

### DISCUSSION THREAD

**Discussion by Dexus** [DeineAgenturUG/greenbone-gvm-openvas-for-docker#31] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by harshalgithub** [comment]
Hi Team,

Great @Dexus , thanks for creating discussion thread, 

This will surely help community to collaborate and learn , discover more.

**Comment by harshalgithub** [comment]
We will start with very very basic query, so later we can discuss this on more broader way.

1. How you are creating the Docker Image of greenbone-gvm-openvas-for-docker.?

2. What are the steps to create **_separate_** images for GVM, GSA, Openvas using docker from scratch,     as Greenbone is not providing the docker images?  

Thanks in advance.

**Reply by Dexus** [reply]
1. I run the scripts :D and have my own environment only to build the image, with cache of deb files, compiling of missing packages/architectures..

2. https://hub.docker.com/u/greenbone has all images you need to run them as docker compose with each as an separated image.
But this setup, is a bit complicated and in view of performance and needs of sockets, it is absolutely not recommend to separate all the images as of my own thoughts.

**Reply by harshalgithub** [reply]
1. Great. Thanks. Can you share the scripts, if possible? . Do you have architecture diagram for your image?

2. I have just gone through greenbone provided images, but they have not given descriptions for images or how to execute it. (which ports to expose.. etc..)

So from naming conventions, what I understand is below images are required components  to start docker compose ? (Did I missed any image or any extra image below? )

https://hub.docker.com/r/greenbone/gvmd (GVM Daemon)
https://hub.docker.com/r/greenbone/gsad (GSA Daemon)
https://hub.docker.com/r/greenbone/openvas-scanner (Openvas Scanner Image)
https://hub.docker.com/r/greenbone/gsa (GSA admin)
https://hub.docker.com/r/greenbone/redis-server (Redis Server)
https://hub.docker.com/r/greenbone/pg-gvm (Postgres GVM Database)

**Reply by Dexus** [reply]
1. watch to the repository, [local_multistep_build_v2.sh](https://github.com/DeineAgenturUG/greenbone-gvm-openvas-for-docker/blob/main/local_multistep_build_v2.sh). What do you mean with a architecture diagram for my image? Official only x86_64 are supported.

2. the name conversation from https://hub.docker.com is the same as for the https://github.com/greenbone/ repositories, more or less. Maybe not all images can found public or use the public images only with a tweaked config that you also find in the other repositories.

**Comment by NekoWLK** [comment]
@Dexus First, I want to thank you for your valuable efforts on this project. 

Second, I have a question, is it possible to migrate the data that is in an old installation of version 21.04.0 to the current version of the image you have in your repository?

Thank you very much for your answer.

**Reply by Dexus** [reply]
Hi @NekoWLK 

sorry for the late answer, yes it should work. The question is from which image and version, so I can give you more help.

If you migrate from an old Secure-Compliance-Solutions Image based on Debian, you should be able to migrate.

Have a look to our WIKI
- first: https://github.com/DeineAgenturUG/greenbone-gvm-openvas-for-docker/wiki/Database-upgrade-from-version-below-Postgresql-13 
- then you can setup you GVM Container as described in our wiki with the migrated DB.

**Reply by NekoWLK** [reply]
@Dexus, thanks for your answer. 

I have a container based on Secure-Compliance-Solutions image (version 21.4.0) with a configurated volume. 

Thanks!

**Reply by Dexus** [reply]
Only one volume? That is mounted to `/data`?  Than we can't migrate this currently.

If you have this, we can't migrate it currently:

Before and equal TAG 21.4.0-v5

```
volumes:
    - ./gvmdata:/data
```
